src/aws-elasticache/user.ts

```diff
@@ -0,0 +1,335 @@
+import {
+  IResource,
+  Lazy,
+  Names,
+  Resource,
+  Stack,
+  aws_elasticache,
+  SecretValue,
+  Token,
+  aws_iam,
+  ArnFormat,
+} from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { Engine } from './util';
+
+/**
+ * An User
+ */
+export interface IUser extends IResource {
+  /**
+   * The ARN of the user.
+   *
+   * @attribute
+   */
+  readonly userArn: string;
+  /**
+   * The ID of the user.
+   *
+   * @attribute
+   */
+  readonly userId: string;
+
+  /**
+   * Grant the given identity the specified actions
+   */
+  grant(grantee: aws_iam.IGrantable, ...actions: string[]): aws_iam.Grant;
+
+  /**
+   * Grant the given identity connection access to the cache.
+   */
+  grantConnect(grantee: aws_iam.IGrantable): aws_iam.Grant;
+}
+
+/**
+ * Specifies the authentication type.
+ */
+export enum AuthenticationType {
+  /**
+   * Password required.
+   */
+  PASSWORD = 'password',
+
+  /**
+   * No password required
+   */
+  NO_PASSWORD_REQUIRED = 'no-password-required',
+
+  /**
+   * IAM
+   */
+  IAM = 'iam',
+}
+
+/**
+ * Properties for defining an User.
+ */
+export interface UserProps {
+  /**
+   * The username of the user.
+   *
+   * The name can have up to 120 characters, and must not contain spaces.
```

Comment:
This constraint is quoted from the management console. It is not written in the [CFn documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticache-user.html#cfn-elasticache-user-username).

<img width="605" alt="image" src="https://github.com/user-attachments/assets/c2ce9c51-fa2d-4b8e-b45d-6339385b70ba">